uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -66,6 +67,8 @@ def connection_made(self, transport):
     def connection_lost(self, exc):
         self.connections.remove(self)
         self.handshake_completed_event.set()
+        if self.task is not None:
+            self.task.cancel()
```

Comment:
I'm not sure we should just cancel the task here.

The web application needs to be notified via ASGI that the socket has shutdown, but it does need to be able to run anything else to completion without having the task cancelled at an arbitrary point in it's execution.

I reckon we'll want to think about some of this in more detail.